From 6ac9562d756d42592994aeaac7a5e7b2bbce5cab Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sat, 14 Dec 2019 14:07:57 -0800
Subject: [PATCH] librdmacm: Use sched_yield instead of pthread_yield

glibc redirects pthread_yield to sched_yield additionally we get it
working with musl on linux

Upstream-Status: Submitted [https://github.com/linux-rdma/rdma-core/pull/635]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 librdmacm/rsocket.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/librdmacm/rsocket.c b/librdmacm/rsocket.c
index 89ae2139..7fc5b266 100644
--- a/librdmacm/rsocket.c
+++ b/librdmacm/rsocket.c
@@ -3038,7 +3038,7 @@ static int rs_poll_enter(void)
 	pthread_mutex_lock(&mut);
 	if (suspendpoll) {
 		pthread_mutex_unlock(&mut);
-		pthread_yield();
+		sched_yield();
 		return -EBUSY;
 	}
 
-- 
2.24.1

